corehq/apps/users/tests/test_username_validation.py

```diff
@@ -0,0 +1,135 @@
+from django.test import SimpleTestCase, TestCase
+
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.users.exceptions import (
+    InvalidDomainException,
+    InvalidUsernameException,
+    UsernameAlreadyExists,
+)
+from corehq.apps.users.models import CommCareUser
+from corehq.apps.users.validation import (
+    ReservedUsernameException,
+    _check_for_reserved_usernames,
+    _ensure_username_is_available,
+    _ensure_valid_username,
+    validate_mobile_username,
+)
+
+
+class TestMobileUsernameValidation(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = 'test-domain'
+        cls.domain_obj = create_domain('test-domain')
+        cls.addClassCleanup(cls.domain_obj.delete)
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.user = CommCareUser.create(self.domain, 'test-user@test-domain.commcarehq.org', 'abc123', None, None)
```

Comment:
does this need to be in `setUp`? It will get deleted and re-created for each test case. I don't see any need for that so I think it could be moved to the `setUpClass`

Reply:
See e9bb80ded2dd47c3a2f1c608fe2545524ed91b59